### ISSUE TITLE
Launchbox Support - Manual Location Override

### ISSUE THREAD

**Issue by ecleese** [mmatyas/pegasus-frontend#490]
Is there a way to manually set the location of Launchbox? I don't have my Launchbox installation in the default location and Pegasus can't find the installation location.

**Comment by mmatyas** [comment]
Not *yet* but I think I'll be able to add support for it next week.

**Comment by mmatyas** [comment]
Actually I just had some time today and added this feature. There's no UI interface for it, but you can add the following line to your settings file:
```
providers.launchbox.installdir: C:/path/to/Launchox
```
It wasn't 100% tested so feel free to report if it fails or something. The settings file is located at `C:/Users/[username]/AppData/Local/pegasus-frontend/settings.txt`.

**Comment by ecleese** [comment]
Awesome! I was able to load the content after adding a trailing "/" to the install directory line however now I receive an error when trying to launch any game due to no launch command defined:

`2019-08-02T15:41:30 [w] Cannot launch the game `Ballistix` because there is no launch command defined for it.`

All of the games I tried work from LaunchBox fine. Are there additional settings I should be configuring for launching or should that all be part of the native LaunchBox integration?

**Comment by mmatyas** [comment]
That might be a bug. Will look into it tomorrow.

**Comment by mmatyas** [comment]
Hm I've just tried it with my LaunchBox installation but everything seemed to work fine. Maybe there's something different compared to my files; could you upload your `Data` directory?

**Comment by ecleese** [comment]
Hopefully later in the day I can get the directory contents together for you. One thing I noticed in the lastrun.log file is any item that has an absolute file path isn't recognized by Pegasus. Any relative path to the root folder of LaunchBox works. This appears to be the case for emulators as well as games. Here is an example:

2019-08-04T09:54:52 [w] LaunchBox: emulator `E:/LaunchBox/E:/Games/Emulators/snes9x/snes9x-x64.exe` doesn't seem to exist, entry ignored


It's been on my todo list to clean up my file locations and make all contents relative to LaunchBox's root folder structure but others may use absolute paths in too. Is it possible to add support for absolute paths?

**Comment by mmatyas** [comment]
Oh I see, it's the absolute path support that was missing. Will fix it real quick!

**Comment by ecleese** [comment]
Absolute paths are now working, nice job! 

I am running into another issue when Pegasus tries to launch RetroArch games. Regardless of the game's platform, Pegasus is always specifying `fbneo_libretro.dll` as the core type. One example:

`2019-08-04T15:33:16 [i] Executing command: [`E:/Emulators/RetroArch/retroarch.exe`,`-L`,`cores\fbneo_libretro.dll`,`E:\Games\genesis\Beyond Oasis.bin`]`

The core specified in my LaunchBox config for my Genesis platform is `genesis_plus_gx_libretro.dll` and LaunchBox does use this core when starting the game.

In addition to the incorrect core, the command doesn't seem to understand the relative core path and errors out stating:

```
[WARN] --libretro argument "cores\fbneo_libretro.dll" is neither a file nor directory. Ignoring.
[ERROR] RetroArch is built for dynamic libretro cores, but libretro_path is not set. Cannot continue
```

I also had trouble getting the game to launch with commas in the command. This is the full command that worked for me after changing the core type, adding the full core path, removing commas and single quotes and adding double quotes to the core and game paths. 

`E:/Emulators/RetroArch/retroarch.exe -L "E:\Emulators\RetroArch\cores\genesis_plus_gx_libretro.dll" "E:\Games\genesis\Beyond Oasis.bin"`

Hopefully this helps you find what to change. All other emulators I've tested so far are launching correctly.

**Comment by mmatyas** [comment]
Hm not sure yet what could cause this. The `Data` directory would really help :)

**Comment by ecleese** [comment]
Here you go. I only included the Genesis platform xml as an example platform. [Data.zip](https://github.com/mmatyas/pegasus-frontend/files/3465317/Data.zip)

**Comment by mmatyas** [comment]
Thanks!

**Comment by mmatyas** [comment]
Turned out there can be various cases of emulator settings other than what was present in my small setup. Had to do some rewriting, but the launch command themselves *should* be correct now.

(The `Executing command` part is only for logging purposes, the `,`s are not actually part of the command.)

**Comment by ecleese** [comment]
Launching is working great now, thanks. 

Any idea what would cause RetroArch not to quit via ESC key when Pegasus launches it? If I run RetroArch on its own or through LaunchBox the ESC key closes RetroArch no problem. I checked the RetroArch key config and it is set to ESC for "quit". I also tried deleting Esc from the following settings line but the problem persists:

`keys.cancel: Backspace,GamepadB`

Other emulators close fine with the ESC key after Pegasus launches them, it only seems isolated to RetroArch.
